### PR TITLE
Test refactorings

### PR DESCRIPTION
**Pull request by stejskalleos**
- Use Github actions instead of Jenkins
- Fix 6.1 routing issues in tests

Jenkins removal PR: https://github.com/theforeman/jenkins-jobs/pull/225

### CODE REVIEW

**Comment by Ron-Lavi** [comment]
ok to test

**Comment by stejskalleos** [comment]
[test test_ruby]

File: .github/workflows/ruby_ci.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  DATABASE_CLEANER_ALLOW_REMOTE_DATABASE_URL: true
+jobs:
+  rubocop:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.5
```
Comment (Ron-Lavi) [inline .github/workflows/ruby_ci.yaml:19]: Should it be 2.7?

File: .github/workflows/ruby_ci.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      postgres:
+        image: postgres:12.1
+        ports: ['5432:5432']
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+    strategy:
+      fail-fast: false
+      matrix:
+        foreman-core-branch: [develop]
+        ruby-version: [2.5]
+        node-version: [12]
```
Comment (Ron-Lavi) [inline .github/workflows/ruby_ci.yaml:40]: not sure if we need this matrix, but we can keep it,
should we also update to Ruby 2.7?

File: config/routes/mount_engine.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 Foreman::Application.routes.draw do
-  mount ForemanBootdisk::Engine, at: '/bootdisk'
+  mount ForemanBootdisk::Engine, at: '/', as: 'foreman_bootdisk'
```
Comment (Ron-Lavi) [inline config/routes/mount_engine.rb:4]: didn't test it yet, though would it break existing API calls or hard-coded routes to `/bootdisk/*`?

File: .github/workflows/ruby_ci.yaml (hunk 1 above)
Comment (stejskalleos) [reply .github/workflows/ruby_ci.yaml:19]: Nope, we still have to support 2.5

File: config/routes/mount_engine.rb (hunk 3 above)
Comment (stejskalleos) [reply config/routes/mount_engine.rb:4]: If you change it to `/bootdisk` then test will fail:

```
be rails test:foreman_bootdisk

Failure:
ForemanBootdisk::Api::V2::DisksControllerTest::default API version 2#test_0001_path - /api/hosts/:host_id routes to #host [/home/lstejskal/projects/foreman_bootdisk/test/functional/foreman_bootdisk/api/v2/disks_controller_test.rb:81]
Minitest::Assertion: The generated path </bootdisk/api/v2/hosts/443> did not match </api/v2/hosts/443>.
Expected: "/api/v2/hosts/443"
  Actual: "/bootdisk/api/v2/hosts/443"

Failure:
ForemanBootdisk::Api::V2::DisksControllerTest::default API version 2#test_0002_path - /api/generic routes to #generic [/home/lstejskal/projects/foreman_bootdisk/test/functional/foreman_bootdisk/api/v2/disks_controller_test.rb:90]
Minitest::Assertion: The generated path </bootdisk/api/generic> did not match </api/generic>.
Expected: "/api/generic"
  Actual: "/bootdisk/api/generic"

66 tests, 151 assertions, 2 failures, 0 errors, 0 skips

```

Comment (Ron-Lavi) [reply config/routes/mount_engine.rb:4]: though it makes sense to me,
I saw this pattern across multiple plugins: https://github.com/search?q=org%3Atheforeman+engine_name&type=code
```rb
module ForemanPlugin
  class Engine < ::Rails::Engine
  engine_name 'foreman_plugin'
  ...
  ...
```

Comment (Ron-Lavi) [reply config/routes/mount_engine.rb:4]: here's also an example PR of how it was added in foreman_tasks: https://github.com/theforeman/foreman-tasks/commit/00cc6abfed25dcdba0ee56dbfe89de1d6af9128f

**Comment by stejskalleos** [comment]
Moving back to draft, there are still some issues, this time with helpers ...

**Review by Ron-Lavi — APPROVED** [review]
Thanks @stejskalleos

**Pull request merged**